features/steps/ruby_notifier_steps.rb

```diff
@@ -12,4 +12,30 @@
   steps %Q{
     When I set environment variable "#{env_var}" to "#{credentials}@#{current_ip}:#{MOCK_API_PORT}"
   }
+end
+Then(/^the "(.+)" of the top project stackframe equals "(.+)"(?: for request (\d+))?$/) do |element, value, request_index|
+  stacktrace = read_key_path(find_request(request_index)[:body], 'events.0.exceptions.0.stacktrace')
+  frame_index = 0
+  stacktrace.each do |frame, index|
+    unless /.*lib\/bugsnag.*\.rb/.match(frame["file"])
+      frame_index = index
+      break
+    end
+  end
+  steps %Q{
+    the "#{element}" of stack frame #{frame_index} equals "#{value}"
+  }
+end
+Then(/^the "(.+)" of the top project stackframe equals (\d+)(?: for request (\d+))?$/) do |element, value, request_index|
```

Comment:
I don't think you need 2 very similar step definitions; you can probably get away with using `(\d+|(".+"))` (so the double quotes are part of the capture for the string case and you can just have `equals #{value}`)